### PR TITLE
Fix away.js logging all incoming Messages

### PR DESCRIPTION
**Pull request by huffman**

### CODE REVIEW

**Comment by huffman** [comment]
To test:

 1. Make sure away.js still works
 1. Make sure "away.js | Got message on Hifi-Away-Enable: message" doesn't show up in the logs when you run `Messages.sendLocalMessage('some-channel', 'message')`

File: scripts/system/away.js [hunk 1]
```diff
@@ -304,9 +304,9 @@ function setEnabled(value) {
 
 var CHANNEL_AWAY_ENABLE = "Hifi-Away-Enable";
 var handleMessage = function(channel, message, sender) {
-    print("Got away message");
-    if (channel == CHANNEL_AWAY_ENABLE) {
-        setEnabled(message == 'enable');
+    if (channel === CHANNEL_AWAY_ENABLE) {
```
Comment (ZappoMan) [inline scripts/system/away.js]: jeez -- whoever designed this messaging API sure made it hard on script developers

**Comment by hifi-gustavo** [comment]
Build console output for commit: 727fabeab2a48666f07096c038a4b334262fb9ab available at: https://gist.github.com/03bc5df2f4831ae28f1e021fe0c94446<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8907-727fabe.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8907-727fabe.dmg

**Comment by ZappoMan** [comment]
testing now

File: scripts/system/away.js (hunk 1 above)
Comment (howard-stearns) [inline scripts/system/away.js]: Isn't it worse than that? Don't we also need to check that (MyAvatar.sessionUUID === senderID)?
(Make sure to test with multiple users.)

**Comment by ZappoMan** [comment]
PASS

File: scripts/system/away.js [hunk 2]
```diff
@@ -343,6 +343,7 @@ Script.scriptEnding.connect(function () {
     Controller.disableMapping(eventMappingName);
     Controller.mousePressEvent.disconnect(goActive);
     Controller.keyPressEvent.disconnect(maybeGoActive);
+    Messages.messageReceived.disconnect(handleMessage);
```
Comment (howard-stearns) [inline scripts/system/away.js]: Do we need a Messages.unsubscribe(CHANNEL_AWAY_ENABLE) ?

File: scripts/system/away.js (hunk 1 above)
Comment (ZappoMan) [inline scripts/system/away.js]: this PR just moves the print message into the channel filter - if we were sending people into away mode there would be much more obvious complaints/problems

Comment (huffman) [inline scripts/system/away.js]: Technically it's vulnerable to it, but no we aren't doing that. I can create another PR with that change

**Pull request merged**